src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNegotiatedCipherSuiteTest.cs

```diff
@@ -635,32 +637,42 @@ private static async Task<Exception> WaitForSecureConnection(VirtualNetwork conn
             // Now we expect both sides to fail or both to succeed
 
             Exception failure = null;
+            Task task = null;
 
             try
             {
-                await serverTask.ConfigureAwait(false);
+                task = await Task.WhenAny(serverTask, clientTask).TimeoutAfter(TestConfiguration.PassingTestTimeoutMilliseconds).ConfigureAwait(false);
+                await task ;
```

Comment:
We already have a helper for this:
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs#L64

Reply:
WhenAllOrAnyFailed() does not give me task which finished. In this case one can finish possibly with error and I need to closer stream just to kick the other half out.  (at least for the negative test  cases)